src/SAML2/Compat/AbstractContainer.php

```diff
@@ -92,6 +97,8 @@ public function registerExtensionHandler(string $class): void
         Assert::subclassOf($class, AbstractXMLElement::class);
         if (is_subclass_of($class, CustomIdentifierInterface::class, true)) {
             $key = $class::getXsiType() . ':BaseID';
+        } elseif (is_subclass_of($class, Condition::class, true)) {
+            $key = $class::getXsiType();
```

Comment:
Shouldn't this be the same as the case for `BaseID`?

Reply:
Not if we put ssp:CustomCondition in the class constant....  It's something we have to find out at some point that works best